### PR TITLE
cli: Restore _suppress_exceptions

### PR DESCRIPTION
**Pull request by jayvdb**
Also update the self-transpile tests to reflect
that cpp, julia and nim no longer have exceptions
occur due to recent bug fixes.
Only go still has exceptions during self-transpilation.

### CODE REVIEW

**Comment by jayvdb** [comment]
@adsharma, ping, this is the last regression that I believe is blocking https://github.com/adsharma/py2many/issues/356

File: py2many/clike.py [hunk 1]
```diff
@@ -164,10 +164,13 @@ def _cast(self, name: str, to) -> str:
...
         if sys.version_info < (3, 9, 0):
-            if not isinstance(node.slice, ast.Index):
-                raise AstNotImplementedError("Advanced Slicing not supported", node)
-            slice_value = node.slice.value
+            if isinstance(node.slice, ast.Index):
+                slice_value = node.slice.value
+            else:
+                slice_value = node.slice
         else:
+            if isinstance(node.slice, ast.Slice):
```
Comment (adsharma) [inline py2many/clike.py:172]: Left a comment on https://github.com/adsharma/py2many/issues/422 on this topic.

File: py2many/cli.py [hunk 2]
```diff
@@ -352,7 +352,10 @@ def _relative_to_cwd(absolute_path):
 
 
 def _get_output_path(filename, ext, outdir):
-    output_path = outdir / (filename.stem + ext)
+    directory = outdir.joinpath(str(filename.parent))
```
Comment (adsharma) [inline py2many/cli.py:355]: `outdir / filename.parent`

File: py2many/cli.py [hunk 3]
```diff
@@ -367,7 +370,7 @@ def _process_one(
     Returns False if reformatter failed.
     """
     suffix = f".{args.suffix}" if args.suffix is not None else settings.ext
-    output_path = _get_output_path(filename, suffix, outdir)
+    output_path = _get_output_path(pathlib.Path(filename.name), suffix, outdir)
```
Comment (adsharma) [inline py2many/cli.py:373]: is this change necessary? filename is already a Path.

**Review by adsharma — COMMENTED** [review]
The first three diffs in the stack can be merged with minor changes.
Seeking community input for the advanced slicing. Will report back if I hear anything.

File: py2many/cli.py (hunk 3 above)
Comment (jayvdb) [reply py2many/cli.py:373]: filename can be an absolute path, which breaks the algorithm in `_get_output_path`

File: py2many/clike.py (hunk 1 above)
Comment (jayvdb) [reply py2many/clike.py:172]: This change here is only to make v0.3.0 consistent across Python versions.  I expect this will be supported at some stage, but it isnt worth holding up v0.3.0 to implement it.

File: py2many/cli.py [hunk 4]
```diff
@@ -81,7 +81,10 @@ def core_transformers(tree, trees):
 
 
 def _transpile(
-    filenames: List[pathlib.Path], sources: List[str], settings: LanguageSettings
+    filenames: List[pathlib.Path],
+    sources: List[str],
+    settings: LanguageSettings,
+    _suppress_exceptions=Exception,
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply py2many/cli.py:87]: This is a hidden arg, used only to support our self-transpile tests, and becomes unnecessary and can be removed once all languages can parse Python code without throwing exceptions.  We're getting quite close.  To me it is ridiculous to add typing here, especially as there are so many other typing problems to fix.

File: py2many/cli.py (hunk 2 above)
Comment (jayvdb) [reply py2many/cli.py]: done

**Comment by jayvdb** [comment]
ping @adsharma

File: py2many/toposort_modules.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 def module_for_path(path: Path) -> str:
     # strip out .py at the end
-    return ".".join(path.parts)[:-3]
```
Comment (adsharma) [inline py2many/toposort_modules.py:10]: This doesn't do the same thing as before:

```
>>> module_for_path(Path("a/b/c.py"))
'a.b.c'
>>> module_for_path_after(Path("a/b/c.py"))
'a/b/c'
```

File: py2many/cli.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (adsharma) [reply py2many/cli.py:88]: It wasn't so much about internal/external or temporary/long term change. The most surprising thing about it was it could be a bool or a list of exceptions. It could be resolved with a comment, but type annotations work better sometimes.

Other mitigations are possible - split it into two variables or interpret empty list as boolean False etc.

File: py2many/cli.py (hunk 3 above)
Comment (adsharma) [reply py2many/cli.py:374]: `filename.relative_to(filename.parent)` makes the logic clearer

**Review by adsharma — APPROVED** [review]
Minor nits. I'll merge and fix it up.

File: py2many/toposort_modules.py (hunk 5 above)
Comment (jayvdb) [reply py2many/toposort_modules.py:10]: curious that there were no tests failing.

**Comment by adsharma** [comment]
It's used only for dependency management in toposort_modules.py, so both
variants work.
But it's possible that we start using module_for_path() for other stuff in
which case this becomes significant.
